src/utilities/idf/IdfObject.cpp

```diff
@@ -1383,14 +1383,14 @@ namespace detail {
     unsigned n = numFields();
     if (n < min_n) {
       m_fields.resize(min_n);
+      n = min_n;
     }
     // also make sure extensible groups are whole
     if (m_iddObject.properties().extensible) {
-      n = numFields();
       int nExtFields = n - m_iddObject.numFields();
       if (nExtFields > 0) {
         int groupSize = m_iddObject.properties().numExtensible;
-        int nToAdd = nExtFields % groupSize;
+        int nToAdd = (groupSize - nExtFields) % groupSize;
```

Comment:
Fix for #4268 is here.

m_iddObject.numFields() = 2
groupSize = 6
nExtFields =  6 + 4 (missing 2) = 10

Before

```
nToAdd = nExtFields % groupSize = 10 % 6 = 4
```

Now

```
nToAdd = (6 - 10) % 6 = (-4) % 6 = 2
```

| nExtFields | groupSize – nExtFields | nToAdd = (groupSize – nExtFields)<br>         % groupSize |
|------------|------------------------|-----------------------------------------------------------|
| 1          | 5                      | 5                                                         |
| 2          | 4                      | 4                                                         |
| 3          | 3                      | 3                                                         |
| 4          | 2                      | 2                                                         |
| 5          | 1                      | 1                                                         |
| **6**          | 0                      | **0**                                                          |
| 7          | -1                     | 5                                                         |
| 8          | -2                     | 4                                                         |
| 9          | -3                     | 3                                                         |
| 10         | -4                     | 2                                                         |
| 11         | -5                     | 1                                                         |
| **12**         | -6                     | **0**                                                         |
| 13         | -7                     | 5                                                         |
| 14         | -8                     | 4                                                         |
| 15         | -9                     | 3                                                         |
| 16         | -10                    | 2                                                         |
| 17         | -11                    | 1                                                         |
| **18**         | -12                    | **0**                                                      |
| 19         | -13                    | 5                                                         |
| 20         | -14                    | 4                                                         |

Reply:
so this was specific to ASHRAE9012016_OutPatientHealthCare_Denver.idf or just any idf with extensible fields. If the later I'm surprised this hasn't surfaced before. PR looks good to me and ready to merge.

Reply:
Any IDF with extensible fields **where the extensible group was truncated**. We rarely load plain IDF files (we're more into writing them), even less with Debug builds, so it hasn't surfaced before the epJSONTranslator test.